sdks/java/core/src/main/java/org/apache/beam/sdk/runners/TransformHierarchy.java

```diff
@@ -145,7 +145,7 @@ public void finishSpecifyingInput() {
       Node producerNode = getProducer(inputValue);
       PInput input = producerInput.remove(inputValue);
       inputValue.finishSpecifying(input, producerNode.getTransform());
-      checkState(producers.get(inputValue) != null, "Producer unknown for input %s", inputValue);
+      checkState(getProducer(inputValue) != null, "Producer unknown for input %s", inputValue);
```

Comment:
I mean, you can cut out the entire check. `getProducer` will throw if the returned value is `null`, yes?